backend를 docker-compose로 돌립니다.

evelon opened the pull request:
### 개요
backend를 docker-compose로 돌립니다.

### 작업 사항
docker-compose파일에 backend를 추가하고 Dockerfile도 추가했습니다.

### 변경점
docker-compose로 backend 서버를 돌릴 수 있게 되었습니다.

### 목적
docker-compose를 통해 환경변수를 통합관리하기 위해서입니다.
docker-compose를 통하지 않고서 돌리려면 backend 폴더 안에 따로 .env 파일을 넣어서 실행시키면 됩니다.

closes #8

DiceMono reviewed (approved):
코드 확인했습니다.
나중에 주석 제거해주시고 jwt 토큰을 쿠키에 저장하는법, 가드에서 쿠키에 저장된 토큰을 가져오는 법 등을 적용하면 완벽할 것 같습니다.

evelon committed:
fix: call uncalled procedure category_reseting

evelon committed:
fix: #03 change COUNT to MAX when counting info

evelon committed:
feat: #6 apply swagger-ui & swagger-jsdoc

swagger-jsdoc은 각 코드 파일에 작성된 openapi를 모아줍니다. swagger-ui는
이렇게 만들어진 swagger.yaml파일을 문서로 만들어 배포합니다.
현재 /swagger경로로 들어가면 문서가 나오도록 설정되어있습니다.

evelon committed:
feat: run backend by docker-compose

도커 컴포즈로 백엔드가 돌아갑니다. 3000번 포트로 열어놓았습니다.

naranghae reviewed (approved):
확인했습니다
